src/prefect/infrastructure/process.py

```diff
@@ -121,6 +143,24 @@ async def run(
             status_code=process.returncode, identifier=str(process.pid)
         )
 
+    async def kill(self, infrastructure_pid: str, grace_seconds: int):
+        hostname, pid = _parse_infrastructure_pid(infrastructure_pid)
+
+        if hostname != socket.gethostname():
+            # The process is running on a different host.
+            return
+
+        def send_sigkill():
+            try:
+                os.kill(pid, signal.SIGKILL)
+            except ProcessLookupError:
+                # The process likely exited due to the SIGTERM.
+                pass
+
+        os.kill(pid, signal.SIGTERM)
+        loop = asyncio.get_running_loop()
+        loop.call_later(grace_seconds, send_sigkill)
```

Comment:
Hmm, this should probably block until the process is killed so instead we should `await anyio.sleep(grace_seconds)` then send the SIGKILL? The agent can submit it to a task group to avoid blocking. We may need a polling interval to actually watch for the kill. You'd want to return early on that `ProcessLookupError`